### PR TITLE
Antagus Minor:Demonus

### PR DESCRIPTION
**Pull request by Aurorablade**
First in a series of mini antag updates.

:cl: Fethas
rscadd:Bloodcrawl is now a spell. The demon heart now properly adds and removes this spell for humans.
tweak:You hopefully will not be able to use robot gibs to blood crawl.
/:cl:

### CODE REVIEW

**Comment by tigercat2000** [comment]
conflicts already? :wink:

**Comment by Aurorablade** [comment]
tis was an old branch....before i started doing some organ fixes....

File: code/datums/spells/bloodcrawl.dm [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+	desc = "Use pools of blood to phase out of existence."
+	charge_max = 0
+	clothes_req = 0
+	selection_type = "range"
+	range = 1
+	cooldown_min = 0
+	overlay = null
+	action_icon_state = "bloodcrawl"
+	action_background_icon_state = "bg_demon"
+	panel = "Daemon"
```
Comment (tigercat2000) [inline code/datums/spells/bloodcrawl.dm]: H E R E S Y

File: code/game/gamemodes/miniantags/slaughter/bloodcrawl.dm [hunk 2]
```diff
@@ -19,84 +19,84 @@
...
-					sleep(30)
-				if (kidnapped)
-					src << "<B>You devour [kidnapped]. Your health is fully restored.</B>"
-					src.adjustBruteLoss(-1000)
-					src.adjustFireLoss(-1000)
-					src.adjustOxyLoss(-1000)
-					src.adjustToxLoss(-1000)
-
-					if (istype(src, /mob/living/simple_animal/slaughter)) //rason, do not want humans to get this
+	src.notransform = TRUE
```
Comment (tigercat2000) [inline code/game/gamemodes/miniantags/slaughter/bloodcrawl.dm]: `src.`

File: code/game/gamemodes/miniantags/slaughter/bloodcrawl.dm [hunk 3]
```diff
@@ -19,84 +19,84 @@
...
+	spawn(0)
+		src.visible_message("<span class='danger'>[src] sinks into [B].</span>")
```
Comment (tigercat2000) [inline code/game/gamemodes/miniantags/slaughter/bloodcrawl.dm]: `src.`

File: code/game/gamemodes/miniantags/slaughter/bloodcrawl.dm [hunk 4]
```diff
@@ -19,84 +19,84 @@
...
+		playsound(get_turf(src), 'sound/misc/enter_blood.ogg', 100, 1, -1)
+		var/obj/effect/dummy/slaughter/holder = new /obj/effect/dummy/slaughter( mobloc )
```
Comment (tigercat2000) [inline code/game/gamemodes/miniantags/slaughter/bloodcrawl.dm]: these `( mobloc )` spaces are heresy

File: code/game/gamemodes/miniantags/slaughter/bloodcrawl.dm [hunk 5]
```diff
@@ -19,84 +19,84 @@
...
-						var/mob/living/simple_animal/slaughter/demon = src
-						demon.devoured++
-						kidnapped << "<span class='userdanger'>You feel teeth sink into your flesh, and the--</span>"
-						kidnapped.adjustBruteLoss(1000)
-						kidnapped.forceMove(src)
-						demon.consumed_mobs.Add(kidnapped)
-					else
-						kidnapped.ghostize()
-						qdel(kidnapped)
+		src.ExtinguishMob()
```
Comment (tigercat2000) [inline code/game/gamemodes/miniantags/slaughter/bloodcrawl.dm]: `src.`

File: code/game/gamemodes/miniantags/slaughter/bloodcrawl.dm [hunk 6]
```diff
@@ -19,84 +19,84 @@
...
+		if(src.buckled)
```
Comment (tigercat2000) [inline code/game/gamemodes/miniantags/slaughter/bloodcrawl.dm]: `src.`

File: code/game/gamemodes/miniantags/slaughter/bloodcrawl.dm [hunk 7]
```diff
@@ -19,84 +19,84 @@
...
+			src.buckled.unbuckle_mob()
```
Comment (tigercat2000) [inline code/game/gamemodes/miniantags/slaughter/bloodcrawl.dm]: `src.`

File: code/game/gamemodes/miniantags/slaughter/bloodcrawl.dm [hunk 8]
```diff
@@ -19,84 +19,84 @@
...
+		if(src.pulling && src.bloodcrawl == BLOODCRAWL_EAT)
```
Comment (tigercat2000) [inline code/game/gamemodes/miniantags/slaughter/bloodcrawl.dm]: `src.` `src.`

File: code/game/gamemodes/miniantags/slaughter/bloodcrawl.dm [hunk 9]
```diff
@@ -19,84 +19,84 @@
...
+			if(istype(src.pulling, /mob/living/))
```
Comment (tigercat2000) [inline code/game/gamemodes/miniantags/slaughter/bloodcrawl.dm]: :eyes:

File: code/game/gamemodes/miniantags/slaughter/bloodcrawl.dm [hunk 10]
```diff
@@ -19,84 +19,84 @@
...
+				var/mob/living/victim = src.pulling
```
Comment (tigercat2000) [inline code/game/gamemodes/miniantags/slaughter/bloodcrawl.dm]: :eyes:

File: code/game/gamemodes/miniantags/slaughter/bloodcrawl.dm [hunk 11]
```diff
@@ -19,84 +19,84 @@
...
+				if(victim.stat == CONSCIOUS)
+					src.visible_message("<span class='warning'>[victim] kicks free of [B] just before entering it!</span>")
```
Comment (tigercat2000) [inline code/game/gamemodes/miniantags/slaughter/bloodcrawl.dm]: :eyes:

File: code/game/gamemodes/miniantags/slaughter/bloodcrawl.dm [hunk 12]
```diff
@@ -19,84 +19,84 @@
...
 				else
-					src << "<span class='danger'>You happily devour... nothing? Your meal vanished at some point!</span>"
+					victim.loc = holder///holder
+					victim.emote("scream")
+					src.visible_message("<span class='warning'><b>[src] drags [victim] into [B]!</b></span>")
```
Comment (tigercat2000) [inline code/game/gamemodes/miniantags/slaughter/bloodcrawl.dm]: :eyes:

File: code/game/gamemodes/miniantags/slaughter/bloodcrawl.dm [hunk 13]
```diff
@@ -19,84 +19,84 @@
...
+		if(src.buckled)
+			src.buckled.unbuckle_mob()
+		if(src.pulling && src.bloodcrawl == BLOODCRAWL_EAT)
+			if(istype(src.pulling, /mob/living/))
+				var/mob/living/victim = src.pulling
+				if(victim.stat == CONSCIOUS)
+					src.visible_message("<span class='warning'>[victim] kicks free of [B] just before entering it!</span>")
 				else
-					src << "<span class='danger'>You happily devour... nothing? Your meal vanished at some point!</span>"
+					victim.loc = holder///holder
```
Comment (tigercat2000) [inline code/game/gamemodes/miniantags/slaughter/bloodcrawl.dm]: forceMove() mobs always

File: code/game/gamemodes/miniantags/slaughter/bloodcrawl.dm [hunk 14]
```diff
@@ -19,84 +19,84 @@
...
+					victim.emote("scream")
+					src.visible_message("<span class='warning'><b>[src] drags [victim] into [B]!</b></span>")
+					kidnapped = victim
+		flick("jaunt",animation)
+		src.loc = holder
```
Comment (tigercat2000) [inline code/game/gamemodes/miniantags/slaughter/bloodcrawl.dm]: :eyes:

File: code/game/gamemodes/miniantags/slaughter/bloodcrawl.dm [hunk 15]
```diff
@@ -19,84 +19,84 @@
...
+		src.holder = holder
```
Comment (tigercat2000) [inline code/game/gamemodes/miniantags/slaughter/bloodcrawl.dm]: :eyes:

File: code/game/gamemodes/miniantags/slaughter/bloodcrawl.dm [hunk 16]
```diff
@@ -19,84 +19,84 @@
...
+
+		if(kidnapped)
+			src << "<B>You begin to feast on [kidnapped]. You can not move while you are doing this.</B>"
+			src.visible_message("<span class='warning'><B>Loud eating sounds come from the blood...</B></span>")
+			sleep(6)
+			if (animation)
```
Comment (tigercat2000) [inline code/game/gamemodes/miniantags/slaughter/bloodcrawl.dm]: no spaces between if and (

File: code/game/gamemodes/miniantags/slaughter/bloodcrawl.dm [hunk 17]
```diff
@@ -19,84 +19,84 @@
...
+				qdel(animation)
+			for(var/i = 3; i > 0; i--)
+				playsound(get_turf(src),'sound/misc/Demon_consume.ogg', 100, 1)
+				sleep(30)
+			if (kidnapped)
```
Comment (tigercat2000) [inline code/game/gamemodes/miniantags/slaughter/bloodcrawl.dm]: no spaces between if and (

File: code/game/gamemodes/miniantags/slaughter/bloodcrawl.dm [hunk 18]
```diff
@@ -19,84 +19,84 @@
...
+				src << "<B>You devour [kidnapped]. Your health is fully restored.</B>"
+				src.adjustBruteLoss(-1000)
```
Comment (tigercat2000) [inline code/game/gamemodes/miniantags/slaughter/bloodcrawl.dm]: :eyes:

File: code/game/gamemodes/miniantags/slaughter/bloodcrawl.dm [hunk 19]
```diff
@@ -19,84 +19,84 @@
...
+				src.adjustFireLoss(-1000)
```
Comment (tigercat2000) [inline code/game/gamemodes/miniantags/slaughter/bloodcrawl.dm]: :eyes:

File: code/game/gamemodes/miniantags/slaughter/bloodcrawl.dm [hunk 20]
```diff
@@ -19,84 +19,84 @@
...
+				src.adjustOxyLoss(-1000)
```
Comment (tigercat2000) [inline code/game/gamemodes/miniantags/slaughter/bloodcrawl.dm]: :eyes: @

File: code/game/gamemodes/miniantags/slaughter/bloodcrawl.dm [hunk 21]
```diff
@@ -19,84 +19,84 @@
...
+				src.adjustToxLoss(-1000)
```
Comment (tigercat2000) [inline code/game/gamemodes/miniantags/slaughter/bloodcrawl.dm]: :eyes:

File: code/game/gamemodes/miniantags/slaughter/bloodcrawl.dm [hunk 22]
```diff
@@ -19,84 +19,84 @@
...
+
+				if (istype(src, /mob/living/simple_animal/slaughter)) //rason, do not want humans to get this
```
Comment (tigercat2000) [inline code/game/gamemodes/miniantags/slaughter/bloodcrawl.dm]: no spaces between if and (

File: code/game/gamemodes/miniantags/slaughter/bloodcrawl.dm [hunk 23]
```diff
@@ -19,84 +19,84 @@
...
-				sleep(6)
-				if (animation)
-					qdel(animation)
-			src.notransform = 0
+				src << "<span class='danger'>You happily devour... nothing? Your meal vanished at some point!</span>"
+		else
+			sleep(6)
+			if (animation)
+				qdel(animation)
+		src.notransform = 0
```
Comment (tigercat2000) [inline code/game/gamemodes/miniantags/slaughter/bloodcrawl.dm]: :eyes:

File: code/game/gamemodes/miniantags/slaughter/slaughter.dm [hunk 24]
```diff
@@ -57,23 +57,26 @@
...
-				//Paradise Port:I added the objective for one spawned like this
-				src.mind.objectives += objective
-				src.mind.objectives += fluffObjective
-				src << "<B>Objective #[1]</B>: [objective.explanation_text]"
-				src << "<B>Objective #[2]</B>: [fluffObjective.explanation_text]"
+	var/obj/effect/proc_holder/spell/bloodcrawl/bloodspell = new
+	AddSpell(bloodspell)
+	if(istype(loc, /obj/effect/dummy/slaughter))
+		bloodspell.phased = 1
+	if(src.mind)
```
Comment (tigercat2000) [inline code/game/gamemodes/miniantags/slaughter/slaughter.dm]: :eyes:

File: code/game/gamemodes/miniantags/slaughter/slaughter.dm [hunk 25]
```diff
@@ -57,23 +57,26 @@
...
+		src << src.playstyle_string
+		src << "<B><span class ='notice'>You are not currently in the same plane of existence as the station. Ctrl+Click a blood pool to manifest.</span></B>"
+		src << 'sound/misc/demon_dies.ogg'
+		src.mind.current.verbs += /mob/living/simple_animal/slaughter/proc/slaughterWhisper
```
Comment (tigercat2000) [inline code/game/gamemodes/miniantags/slaughter/slaughter.dm]: :eyes:

File: code/game/gamemodes/miniantags/slaughter/slaughter.dm [hunk 26]
```diff
@@ -57,23 +57,26 @@
...
+		if(!(vialspawned))
+			var/datum/objective/slaughter/objective = new
+			var/datum/objective/demonFluff/fluffObjective = new
+			ticker.mode.traitors |= src.mind
+			objective.owner = src.mind
+			fluffObjective.owner = src.mind
+			//Paradise Port:I added the objective for one spawned like this
+			src.mind.objectives += objective
```
Comment (tigercat2000) [inline code/game/gamemodes/miniantags/slaughter/slaughter.dm]: :eyes:

File: code/game/gamemodes/miniantags/slaughter/slaughter.dm [hunk 27]
```diff
@@ -57,23 +57,26 @@
...
+			src.mind.objectives += fluffObjective
```
Comment (tigercat2000) [inline code/game/gamemodes/miniantags/slaughter/slaughter.dm]: :eyes:

File: code/game/gamemodes/miniantags/slaughter/slaughter.dm [hunk 28]
```diff
@@ -57,23 +57,26 @@
...
+		bloodspell.phased = 1
+	if(src.mind)
+		src << src.playstyle_string
+		src << "<B><span class ='notice'>You are not currently in the same plane of existence as the station. Ctrl+Click a blood pool to manifest.</span></B>"
+		src << 'sound/misc/demon_dies.ogg'
+		src.mind.current.verbs += /mob/living/simple_animal/slaughter/proc/slaughterWhisper
+		if(!(vialspawned))
+			var/datum/objective/slaughter/objective = new
+			var/datum/objective/demonFluff/fluffObjective = new
+			ticker.mode.traitors |= src.mind
```
Comment (tigercat2000) [inline code/game/gamemodes/miniantags/slaughter/slaughter.dm]: :eyes:

File: code/game/gamemodes/miniantags/slaughter/slaughter.dm [hunk 29]
```diff
@@ -57,23 +57,26 @@
...
+			objective.owner = src.mind
```
Comment (tigercat2000) [inline code/game/gamemodes/miniantags/slaughter/slaughter.dm]: :eyes:

File: code/game/gamemodes/miniantags/slaughter/slaughter.dm [hunk 30]
```diff
@@ -57,23 +57,26 @@
...
+			fluffObjective.owner = src.mind
```
Comment (tigercat2000) [inline code/game/gamemodes/miniantags/slaughter/slaughter.dm]: :eyes:

**Comment by Fox-McCloud** [comment]
`src` is implied

spacing is entirely a preference issue--I'm not going to not accept it because of that.

File: code/datums/spells/bloodcrawl.dm (hunk 1 above)
Comment (tigercat2000) [inline code/datums/spells/bloodcrawl.dm]: @Aurorablade this is heresy via "daemon"
it's DEMON YOU COMMIE

**Pull request merged**